server/keyshare/keyshareserver/server.go

```diff
@@ -110,9 +113,14 @@ func (s *Server) Handler() http.Handler {
 		// Registration
 		router.Post("/client/register", s.handleRegister)
 
-		// Pin logic
-		router.Post("/users/verify/pin", s.handleVerifyPin)
+		// authentication
+		router.Post("/users/verify_start", s.handleStartAuth)
+		router.Post("/users/verify/pin", s.handleVerify)
```

Comment:
Because it made sense to me to have the authentication method in the URL, but the code that I needed to handle both endpoints was identical to such a degree that it was easiest to have a single handler method.

I see two options:
* Merge the two endpoints into a single endpoint, called `/users/verify`, i.e. without a suffix indicating the authentication method. This solves the issue of having two endpoints sharing one handler method, but then which authentication method the client chose would have to be determined from the message POSTed to us by it. That seems a bit too implicit to me, and might also complicate things if we ever want to add a new authentication mechanism to this.
* Keep the two endpoints, declared separately here as they are currently, but with a sanity check within the handler method that if `pin_challengeresponse` is invoked, the challenge-response mechanism is actually used. (An endpoint indication as parameter to `handleVerify` is not necessary; instead `handleVerify can just check `r.URL`.)

Which do you prefer?

Reply:
Second options sounds fine :)